src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java

```diff
@@ -252,4 +252,20 @@ public SampleSummary ingest(final MultipartFile file, final String type) throws
         CTPException.Fault.BAD_REQUEST,
         String.format("No sample units were found for sample summary %s", sampleSummaryId));
   }
+
+  @RequestMapping(value = "/sampleunits", method = RequestMethod.GET)
+  public ResponseEntity<List<SampleUnitDTO>> getSampleUnitsByPostcode(
+      @RequestParam("postcode") String postcode) throws CTPException {
+
+    List<SampleUnit> sampleUnits = sampleService.findSampleUnitsByPostcode(postcode);
+    List<SampleUnitDTO> result = mapperFacade.mapAsList(sampleUnits, SampleUnitDTO.class);
+
+    if (!sampleUnits.isEmpty()) {
+      return ResponseEntity.ok(result);
+    }
+
+    throw new CTPException(
```

Comment:
Doesn't feel right to throw an exception when it's not a exceptional scenario. Might be better to return `ResponseEntity.notFound();`